SpatialGDK/Source/SpatialGDKTests/Examples/SpatialGDKExampleTest.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDefinitions.h"
+
+#include "Misc/ScopeTryLock.h"
+#include "HAL/PlatformFilemanager.h"
+
+#define EXAMPLE_SIMPLE_TEST(TestName) \
+	TEST(SpatialGDKExamples, SimpleExamples, TestName)
+
+#define EXAMPLE_COMPLEX_TEST(TestName) \
+	COMPLEX_TEST(SpatialGDKExamples, ComplexExamples, TestName)
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialGDKExamples, Log, All);
+DEFINE_LOG_CATEGORY(LogSpatialGDKExamples);
+
+// 1. Latent command example
+namespace
+{
+	const double MAX_WAIT_TIME_FOR_SLOW_COMPUTATION = 2.0;
+	const double MIN_WAIT_TIME_FOR_SLOW_COMPUTATION = 1.0;
+	const double COMPUTATION_DURATION = 1.0;
+
+	struct
+	{
+		FCriticalSection Mutex;
+		int Value;
+	} ComputationResult;
```

Comment:
Why is this global? Wouldn't it make more sense to define a struct that tests can make an instance of?

Reply:
It's tricky with latent commands. Instantiating this struct in a test with latent commands will likely crash the Editor. Since when running such tests, execution might leave the scope earlier (thus deleting the ComputationResult instance), than every latent command is terminated. I'll think of better solution and I'm open for suggestions.

Reply:
Can the commands own the data they need? I'd argue that if it requires a mutex then your test is getting pretty complicated too.

Reply:
Or give references to each command and let some final cleanup command actually own things.

Reply:
It's not that straightforward with Latent Commands unfortunately.

Reply:
In what way? Can they not capture or can we not pass around references to the stuff?

Reply:
Updated with unique pointer being passed to functions.